File: 02_Programming_Fundamentals/04_Programming_Fundamentals_PHP/17_Associative_Arrays_Exercise/10_softuni_exam_results.php
<?php

$input = "";
$resultsList = [];
$submissionsList = [];

while (true) {
  $input = readline();
  if ($input == "exam finished") { break; }
  $inputArr = explode("-", $input);

  if ($inputArr[1] != "banned") {
    $user = $inputArr[0];
    $language = $inputArr[1];
    $points = intval($inputArr[2]);

    if (!key_exists($user, $resultsList)) {
      $resultsList[$user] = $points;
    } else {
      if ($resultsList[$user] < $points) {
        $resultsList[$user] = $points;
      }
    }

    if (!key_exists($language, $submissionsList)) {
      $submissionsList[$language] = 0;
    }
    $submissionsList[$language] += 1;

  } else {
    $user = $inputArr[0];
    unset($resultsList[$user]);

  }

}

uksort($resultsList, function ($a, $b) use ($resultsList) { 
    if ($resultsList[$a] < $resultsList[$b]) return 1;
    elseif ($resultsList[$a] > $resultsList[$b]) return -1; 
    else return strcmp($a, $b); 
});

echo "Results: " . PHP_EOL;
foreach ($resultsList as $users => $points) {
  echo $users . " | " . $points . PHP_EOL;
}

uksort($submissionsList, function ($a, $b) use ($submissionsList) { 
    if ($submissionsList[$a] < $submissionsList[$b]) return 1;
    elseif ($submissionsList[$a] > $submissionsList[$b]) return -1; 
    else return strcmp($a, $b); 
});

echo "Submissions: " . PHP_EOL;
foreach ($submissionsList as $languages => $count) {
  echo $languages . " - " . $count . PHP_EOL;
}

?>

File: 02_Programming_Fundamentals/02_Programming_Fundamentals_with_C#/20_Associative_Arrays_Exercise/10_SoftUniExamResults.cs
using System;
using System.Linq;
using System.Collections.Generic;

class MainClass
{
    public static void Main(string[] args)
    {
        var dictExam = new Dictionary<string, int>();
        var dictSubm = new Dictionary<string, int>();

        while (true)
        {
            string[] input = Console.ReadLine().Split("-");
            if (input[0] == "exam finished") { break; }

            if (input.Length == 3) {
                string username = input[0];
                string language = input[1];
                int points = int.Parse(input[2]);

                // User - Points
                if (!dictExam.ContainsKey(username)) {
                    dictExam[username] = 0;
                }
                if (points > dictExam[username]) {
                    dictExam[username] = points;
                }

                // Language - Submissions
                if (!dictSubm.ContainsKey(language))
                {
                    dictSubm[language] = 0;
                }
                dictSubm[language]++;

            } else if (input.Length == 2 && input[1].ToLower() == "banned") {
                string username = input[0];
                dictExam.Remove(username);
            }
        }

        Console.WriteLine("Results:");
        foreach (var item in dictExam.OrderByDescending(x => x.Value).ThenBy(x => x.Key))
        {
            Console.WriteLine(item.Key + " | " + item.Value);
        }
        Console.WriteLine("Submissions:");
        foreach (var item in dictSubm.OrderByDescending(x => x.Value).ThenBy(x => x.Key))
        {
            Console.WriteLine(item.Key + " - " + item.Value);
        }

    }
}